### PR TITLE
Save <player>.dat files (with player's inventory)

### PR DESCRIPTION
**Pull request by rrwr**
Continuation of #380

### CODE REVIEW

**Comment by rom1504** [comment]
it seems test are failing

**Comment by rrwr** [comment]
Looks like there is no settings.worldFolder when running tests

**Comment by Karang** [comment]
> Looks like there is no settings.worldFolder when running tests

This will be fixed by https://github.com/PrismarineJS/flying-squid/pull/409

**Comment by rom1504** [comment]
I reviewed this again.
Ultimately I think we'll want to move this in https://github.com/PrismarineJS/prismarine-provider-anvil but as a first step could you try to put the reading and the writing function in a file that just provides stateless function in lib/ folder, and then use it in plugin ?
That will make testing it much easier, and make the code cleaner

**Comment by rrwr** [comment]
OK, but what do you mean by reading function? The code used in #380?

**Comment by rom1504** [comment]
yes basically this part https://github.com/PrismarineJS/flying-squid/pull/380/files#diff-1dea21774fbc02bf339d7f711987fcbbR56

I think something that would take as input the player uuid and world folder and output an object that is just the keys to set in the player object would make a lot of sense
then you'd just loop over this object to set all these properties in the plugin code

File: src/lib/plugins/login.js [hunk 1]
```diff
@@ -36,46 +31,31 @@ module.exports.server = function (serv, options) {
...
+    Object.keys(playerData.player).forEach(k => { player[k] = playerData.player[k] })
+
     serv.players.push(player)
     serv.uuidToPlayer[player.uuid] = player
     player.loadedChunks = {}
   }
 
   function updateInventory () {
-    playerSavedInventoryItems.forEach((item) => {
+    playerData.inventory.forEach((item) => {
```
Comment (rom1504) [inline src/lib/plugins/login.js:58]: why not do this in addPlayer ? (that way you can make playerData local for addPlayer function)

Comment (rrwr) [reply src/lib/plugins/login.js:58]: It seems like player.inventory.updateSlot doesn't work at the time when addPlayer is called

Comment (rrwr) [reply src/lib/plugins/login.js:58]: player.inventory.updateSlot works only after sendLogin

Comment (rom1504) [reply src/lib/plugins/login.js:58]: ok

**Comment by rrwr** [comment]
It seems that now the tests should not fail (since #409 is already merged), can someone restart checks?

**Comment by rom1504** [comment]
no, it's merged in 1.13 branch, not in master

**Comment by rrwr** [comment]
Maybe it's better to rebase this pull request to 1.13 branch?

**Comment by rom1504** [comment]
no, I just merged it to simplify

just rebase on master

**Comment by rrwr** [comment]
ok

**Comment by rom1504** [comment]
world should be saved only if region option is defined, you need to add an if

**Comment by rrwr** [comment]
You mean that <player>.dat file should save only when something is in player.world.anvil.regions?

**Comment by rom1504** [comment]
no, only when this option is not undefined https://github.com/PrismarineJS/flying-squid/blob/af52d3dca819aeada32838638cd4815bba590630/src/lib/plugins/world.js#L18

**Comment by rom1504** [comment]
thanks !

**Pull request merged**